tests/common/reboot.py

```diff
@@ -223,3 +230,47 @@ def check_reboot_cause(dut, reboot_cause_expected):
     reboot_cause_got = get_reboot_cause(dut)
     logging.debug("dut {} last reboot-cause {}".format(dut.hostname, reboot_cause_got))
     return reboot_cause_got == reboot_cause_expected
+
+
+def check_reboot_cause_history(dut, reboot_type_history_queue):
+    """
+    @summary: Check the reboot cause history on DUT. Can be used with wailt_until
+    @param dut: The AnsibleHost object of DUT.
+    @param reboot_type_history_queue: reboot type queue.
+    e.g.
+    show reboot-cause  history
+    Name                 Cause          Time                             User    Comment
+    -------------------  -------------  -------------------------------  ------  ---------
+    2021_09_09_14_15_13  Power Loss ()  N/A                              N/A     N/A
+    2021_09_09_14_06_17  reboot         Thu 09 Sep 2021 02:05:17 PM UTC  admin   N/A
+    2021_09_09_13_59_11  Watchdog ()    N/A                              N/A     N/A
+    2021_09_09_13_52_13  Power Loss ()  N/A                              N/A     N/A
+    2021_09_09_13_45_18  warm-reboot    Thu 09 Sep 2021 01:44:14 PM UTC  admin   N/A
+    2021_09_09_13_37_58  fast-reboot    Thu 09 Sep 2021 01:37:09 PM UTC  admin   N/A
+    2021_09_09_13_30_52  soft-reboot    Thu 09 Sep 2021 01:30:24 PM UTC  admin   N/A
+    2021_09_09_13_24_17  reboot         Thu 09 Sep 2021 01:23:17 PM UTC  admin   N/A
+    """
+    reboot_cause_history_got = dut.show_and_parse("show reboot-cause history")
+    logging.debug("dut {} reboot-cause history {}. reboot type history queue is {}".format(
+        dut.hostname, reboot_cause_history_got, reboot_type_history_queue))
+
+    logging.debug("Check reboot-cause history title")
+    if reboot_cause_history_got:
+        if not set(REBOOT_CAUSE_HISTORY_TITLE) == set(reboot_cause_history_got[0].keys()):
+            logging.debug("Expected reboot-cause history:{} not match actual reboot-cause history:{}".format(
```

Comment:
should this log always be printed to give users a hint about the failure reason? Same comments to the below two debug logging.

Reply:
Yes, I will change it to error level